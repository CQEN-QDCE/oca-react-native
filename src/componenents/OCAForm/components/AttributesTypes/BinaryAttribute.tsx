import React from 'react';
import {
  Image,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  View,
} from 'react-native';
import type { Attribute } from '../../../types';
import { attributeTypeStyle } from '../../../utils/attributeTypeStyle';

interface BinaryAttributeProps {
  attribute: Attribute;
  shown?: boolean;
  binaryStyle?: StyleProp<any>;
  textStyle?: StyleProp<TextStyle>;
}

export const BinaryAttribute = ({
  attribute,
  shown = false,
  binaryStyle,
  textStyle,
}: BinaryAttributeProps) => {
  const defaultStyle = StyleSheet.create({
    image: {
      height: 150,
      aspectRatio: 1,
      resizeMode: 'contain',
      borderRadius: 10,
      ...binaryStyle,
    },
  });

  return (
    <View style={attributeTypeStyle.textValueContainer}>
      {shown ? (
        <Image style={defaultStyle.image} source={{ uri: attribute.value }} />
      ) : (
        <Text style={[attributeTypeStyle.text, textStyle]}>
          {Array(10).fill('\u2022').join('')}
        </Text>
      )}
    </View>
  );
};
