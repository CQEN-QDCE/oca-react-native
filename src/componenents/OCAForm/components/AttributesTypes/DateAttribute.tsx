import React, { useEffect, useState } from 'react';
import { StyleProp, Text, TextStyle, View } from 'react-native';
import type { Attribute } from '../../types';
import { attributeTypeStyle } from '../../utils/attributeTypeStyle';
import moment from 'moment';

interface DateAttributeProps {
  attribute: Attribute;
  shown: boolean;
  styles: StyleProp<TextStyle>;
}

export const DateAttribute = ({
  attribute,
  shown = false,
  styles,
}: DateAttributeProps) => {
  const [date, setDate] = useState<string>();

  useEffect(() => {
    let dataValue: Date | moment.Moment = new Date(attribute.value);
    if (!dataValue && moment(attribute.value, attribute.format).isValid()) {
      dataValue = moment(attribute.value, attribute.format);
    } else {
      return setDate(attribute.value);
    }
    setDate(moment(dataValue).format(attribute.format));
  }, [attribute.value, attribute.format]);

  return (
    <View style={attributeTypeStyle.textValueContainer}>
      <Text style={[attributeTypeStyle.text, styles]}>
        {shown ? date : Array(10).fill('\u2022').join('')}
      </Text>
    </View>
  );
};
