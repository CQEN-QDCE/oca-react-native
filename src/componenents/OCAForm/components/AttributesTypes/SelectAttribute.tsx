import React from 'react';
import { StyleProp, Text, TextStyle, View } from 'react-native';
import type { Attribute } from '../../types';
import { attributeTypeStyle } from '../../utils/attributeTypeStyle';

interface SelectAttributeProps {
  attribute: Attribute;
  shown: boolean;
  styles: StyleProp<TextStyle>;
}

export const SelectAttribute = ({
  attribute,
  shown = false,
  styles,
}: SelectAttributeProps) => {
  return (
    <View style={attributeTypeStyle.textValueContainer}>
      <Text style={[attributeTypeStyle.text, styles]}>
        {shown ? attribute.value : Array(10).fill('\u2022').join('')}
      </Text>
    </View>
  );
};
