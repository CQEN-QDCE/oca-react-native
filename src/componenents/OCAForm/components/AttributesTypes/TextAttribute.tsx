import React from 'react';
import { StyleProp, Text, TextStyle, View } from 'react-native';
import type { Attribute } from '../../../types';
import { attributeTypeStyle } from '../../../utils/attributeTypeStyle';

interface TextAttributeProps {
  attribute: Attribute;
  shown: boolean;
  styles: StyleProp<TextStyle>;
}

export const TextAttribute = ({
  attribute,
  shown = false,
  styles,
}: TextAttributeProps) => {
  return (
    <View style={attributeTypeStyle.textValueContainer}>
      <Text style={[attributeTypeStyle.text, styles]} numberOfLines={200}>
        {shown ? attribute.value : Array(10).fill('\u2022').join('')}
      </Text>
    </View>
  );
};
