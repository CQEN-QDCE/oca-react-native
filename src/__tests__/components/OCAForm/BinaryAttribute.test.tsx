import renderer from 'react-test-renderer';
import React from 'react';
import { StyleSheet } from 'react-native';
import { BinaryAttribute } from '../../../components/OCAForm/components/AttributesTypes/BinaryAttribute';

const binaryAttribute = {
  name: 'Picture',
  value:
    'iVBORw0KGgoAAAANSUhEUgAAAV4AAAFeAQAAAADlUEq3AAAAAW9yTlQBz6J3mgAAACZJREFUaN7twTEBAAAAwqD1T+1pCaAAAAAAAAAAAAAAAAAAAAC4AT2GAAGWvJzxAAAAAElFTkSuQmCC',
  type: 'Binary',
};

describe('BinaryAttribute', () => {
  it('Given a binary attribute should render the attribute value properly.', () => {
    const tree = renderer
      .create(<BinaryAttribute attribute={binaryAttribute} shown={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Given a hidden attribute with custom styling should render hidden value with custom styling.', () => {
    const styles = StyleSheet.create({
      value: {
        fontSize: 32,
        color: 'red',
      },
    });
    const tree = renderer
      .create(
        <BinaryAttribute
          attribute={binaryAttribute}
          shown={false}
          textStyle={styles.value}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Given a binary attribute with custom styling should render the binary attribute properly.', () => {
    const styles = StyleSheet.create({
      imageStyle: {
        height: 200,
        width: 200,
      },
    });
    const tree = renderer
      .create(
        <BinaryAttribute
          attribute={binaryAttribute}
          shown={true}
          binaryStyle={styles.imageStyle}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
