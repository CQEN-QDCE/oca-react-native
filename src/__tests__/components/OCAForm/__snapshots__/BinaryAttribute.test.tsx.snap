// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BinaryAttribute Given a binary attribute should render the attribute value properly. 1`] = `
<View
  style={
    Object {
      "color": "#333",
      "paddingVertical": 4,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "iVBORw0KGgoAAAANSUhEUgAAAV4AAAFeAQAAAADlUEq3AAAAAW9yTlQBz6J3mgAAACZJREFUaN7twTEBAAAAwqD1T+1pCaAAAAAAAAAAAAAAAAAAAAC4AT2GAAGWvJzxAAAAAElFTkSuQmCC",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 10,
        "height": 150,
        "resizeMode": "contain",
      }
    }
  />
</View>
`;

exports[`BinaryAttribute Given a binary attribute with custom styling should render the binary attribute properly. 1`] = `
<View
  style={
    Object {
      "color": "#333",
      "paddingVertical": 4,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "iVBORw0KGgoAAAANSUhEUgAAAV4AAAFeAQAAAADlUEq3AAAAAW9yTlQBz6J3mgAAACZJREFUaN7twTEBAAAAwqD1T+1pCaAAAAAAAAAAAAAAAAAAAAC4AT2GAAGWvJzxAAAAAElFTkSuQmCC",
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 10,
        "height": 200,
        "resizeMode": "contain",
        "width": 200,
      }
    }
  />
</View>
`;

exports[`BinaryAttribute Given a hidden attribute with custom styling should render hidden value with custom styling. 1`] = `
<View
  style={
    Object {
      "color": "#333",
      "paddingVertical": 4,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Object {
          "color": "red",
          "fontSize": 32,
        },
      ]
    }
  >
    ••••••••••
  </Text>
</View>
`;
